feat(manager/unity3d): add Unity Editor manager (#36487)
renovatebot/renovate: lib/modules/manager/unity3d/readme.md

Manages version numbers in the `ProjectVersion.txt` file located within the `ProjectSettings` folder of a Unity project.

Example path: `/home/user/testUnityProject/ProjectSettings/ProjectVersion.txt`.

> `ProjectVersion.txt` always have two version references in `YAML` syntax; one with revision and another one without it.

Reviewer: ```suggestion
`ProjectVersion.txt` always have two version references in YAML syntax; one with revision and another one without it.
```

> Manages version numbers inside the `ProjectVersion.txt` file inside the `ProjectSettings` folder generated by a Unity game engine project. Example: `/home/user/testUnityProject/ProjectSettings/ProjectVersion.txt`.

Reviewer 2: ```suggestion
Manages version numbers in the `ProjectVersion.txt` file located within the `ProjectSettings` folder of a Unity project. 

Example path: `/home/user/testUnityProject/ProjectSettings/ProjectVersion.txt`.
```

> `ProjectVersion.txt` always have two version references in yml syntax; one with and one without a hash, both are updated:

Reviewer 2: ```suggestion
`ProjectVersion.txt` always have two version references in `YAML` syntax; one with revision and another one without it. 
Both are updated together.
```
